packages/web3torrent/src/utils/magnet.ts

```diff
@@ -1,39 +1,34 @@
 import {defaultTrackers, EmptyTorrent} from '../constants';
 import {RoutePath} from '../routes';
 import {Torrent} from '../types';
+import {useLocation} from 'react-router-dom';
 
-export const parseMagnetURL: (rawMagnetURL: string) => Torrent = (rawMagnetURL = '') => {
-  const emptyTorrentData = {...EmptyTorrent, magnetURI: ''} as Torrent;
-  if (!rawMagnetURL.trim()) {
-    return emptyTorrentData;
-  }
-  const magnetParams = new URLSearchParams(rawMagnetURL.trim().replace(/#magnet:/g, ''));
-  if (!magnetParams.has('xt')) {
-    return emptyTorrentData;
-  }
+export const useQuery = () => {
+  return new URLSearchParams(useLocation().search);
+};
 
-  if (!magnetParams.has('tr')) {
-    defaultTrackers.map(tr => magnetParams.append('tr', tr));
+export const parseURL: (infoHash: string, queryParams: any) => Torrent = (
+  infoHash,
+  queryParams
+) => {
+  if (!infoHash || !infoHash.trim()) {
+    return EmptyTorrent;
   }
+  const name = queryParams.get('name') || queryParams.get('dn') || 'unknown';
+  const length = Number(queryParams.get('length')) || 0;
```

Comment:
`||` works here, but, in general, [nullish coalescing operator (??)](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing) is safer.

Reply:
I like the nullish coalescing (though I need to get used to it, I sometimes forget that I can use it).
In this case,  `||` fits better because the queryParam can be an empty string (e.g. `?name=&length=14200`) and this catches that case too.

Reply:
If `queryParams.get('length')` is an empty string, I _think_ `??` would still work correctly since `Number('')` returns 0.